qualtran/qref_interop/bartiq_demo.ipynb

```diff
@@ -0,0 +1,2214 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<Resource name=\"t\" value=\"2*multiplicity(2, N) + 10*ceiling(log2(N/2 ^ multiplicity(2, N)))\">"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "usp_2_bartiq = qref_to_bartiq(usp_2_qref)\n",
+    "compile_routine(usp_2_bartiq).resources[\"t\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "07d684ae-fa77-4d48-8850-4d394b2ec4e3",
+   "metadata": {},
+   "source": [
+    "## Updating Alias Sampling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "d2bddbd7-fd86-400e-8cfe-b593995f91f3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bloq = StatePreparationAliasSampling.from_probabilities(probs_list).decompose_bloq()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "b1a54903-29a3-4017-84cf-e87d51d53a4a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We export our Alias Sampling bloq to QREF and remove top-level resources so we can re-compute them later.\n",
+    "qref_as_1 = bloq_to_qref(bloq)\n",
+    "qref_as_2 = bloq_to_qref(bloq)\n",
+    "qref_as_1.program.resources = []\n",
+    "qref_as_2.program.resources = []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "f2305c8f-7bf0-44a2-8c2f-d6afd63111ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Before we put the subroutine in the right place, we need to make sure it's named correctly, otherwise \n",
+    "# some of the connections will be invalid.\n",
+    "usp_2_qref.program.name = \"PrepareUniformSuperposition_0\"\n",
+    "qref_as_2.program.children.by_name[\"PrepareUniformSuperposition_0\"] = usp_2_qref.program"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "8fe913dc-a880-432f-bc37-d18316d6da92",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<Resource name=\"t\" value=\"307\">"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bartiq_as_1 = qref_to_bartiq(qref_as_1)\n",
+    "compile_routine(bartiq_as_1).resources[\"t\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "72f0b802-a3cd-4e43-86da-ad1bbe0fe39b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/mstechly/Documents/code/2024-06-26-qualtran-conversion/bartiq/src/bartiq/compilation/_compile.py:117: UserWarning: Found the following issues with the provided routine after the compilation has finished: [\"Input param N found in subroutine: CompositeBloq.PrepareUniformSuperposition_0, which is not among top level params: {'PrepareUniformSuperposition_0.N'}.\"]\n",
+      "  warnings.warn(\n"
```

Comment:
We usually clear the output of all notebooks before committing them. The notebooks are run before generating the docs so that the docs generated on https://qualtran.readthedocs.io/en/latest/index.html have the expected output. 

Can you please clear all output from the notebook (assuming the notebooks runs after qualtran runtime dependencies are updated)